openedx/features/course_experience/static/course_experience/js/Enrollment.js

```diff
@@ -0,0 +1,42 @@
+
+/*
+ * Course Enrollment on the Course Home page
+ */
+export class CourseEnrollment {  // eslint-disable-line import/prefer-default-export
+  /**
+   * Redirect to a URL.  Mainly useful for mocking out in tests.
+   * @param  {string} url The URL to redirect to.
+   */
+  static redirect(url) {
+    window.location.href = url;
+  }
+
+  static refresh() {
+    window.location.reload(false);
+  }
+
+  static createEnrollment(courseId) {
+    const data = JSON.stringify({
+      course_details: { course_id: courseId },
+    });
+    const enrollmentAPI = '/api/enrollment/v1/enrollment';
+    const trackSelection = '/course_modes/choose/';
+
+    return () =>
+      $.ajax(
+        {
+          type: 'POST',
+          url: enrollmentAPI,
+          data,
+          contentType: 'application/json',
+        }).done(() => {
+          CourseEnrollment.refresh();
+        }).fail(() => {
+          CourseEnrollment.redirect(trackSelection + courseId);
```

Comment:
1. I believe that the url encoding is handled when setting the href. 

2. The point with the page handling errors is that most of the time, the error is due to a permissions issue, so it's safer to redirect to the track selection page. Most of the other enrollment flows also silently fail and then forward along to the track selection page, so at least this is consistent.